vignettes/mmrm_review_methods.Rmd

```diff
@@ -0,0 +1,582 @@
+the FEV dataset 100 times. The convergence times are recorded for each replicate
+and are reported in the table below.
+
+Table: Comparison of convergence times (milliseconds)
+
+| Implementation | Median | Lower Quartile | Upper Quartile |
+|:---------------|-------:|---------------:|---------------:|
+| `PROC MIXED`   |  20.00 |          20.00 |          20.00 |
+| `mmrm`         |  50.80 |          50.48 |          51.45 |
+| `lmer`         |  97.35 |          96.81 |          97.83 |
+| `glmmTMB`      | 273.45 |         271.77 |         275.28 |
+| `gls`          | 342.08 |         340.57 |         344.04 |
+
+It is clear from these results that `mmrm` converges significantly faster than
+other R functions. Though not demonstrated here, this is generally true
+regardless of the sample size and covariance structure used. `PROC MIXED` is the
+fastest procedure on average, however.
+
+## Marginal Treatment Effect Estimates Comparison
+
+We next estimate the marginal mean treatment effects for each visit using the
+fitted MMRMs. All R implementations' estimates are reported relative to `PROC
+MIXED`'s estimates.
+
+```{r, echo = FALSE, warning = FALSE, message = FALSE}
+library(dplyr)
+library(mmrm)
+library(lme4)
+library(nlme)
+library(glmmTMB)
+library(emmeans)
+library(sasr)
+library(ggplot2)
+
+set.seed(612313)
+
+# compute marginal estimates for mmrm
+mmrm_fit <- mmrm(
+    formula = FEV1 ~ RACE + ARMCD * AVISIT + us(AVISIT | USUBJID),
+    data = fev_data
+  )
+mmrm_ests <- as.data.frame(
+  emmeans(
+    mmrm_fit,
+    spec = trt.vs.ctrl ~ ARMCD | AVISIT,
+    weights = "proportional"
+  )$contrasts
+)$estimate
+
+# compute marginal estimates for lmer
+lmer_fit <- lmer(
+  FEV1 ~ RACE + ARMCD * AVISIT + (0 + AVISIT | USUBJID),
+  data = fev_data[complete.cases(fev_data), ],
+  control = lmerControl(check.nobs.vs.nRE = "ignore")
+)
+lmer_ests <- as.data.frame(
+  emmeans(
+    lmer_fit,
+    spec = trt.vs.ctrl ~ ARMCD | AVISIT,
+    data = fev_data[complete.cases(fev_data), ],
+    weights = "proportional"
+  )$contrasts
+)$estimate
+
+# compute marginal estimates for gls
+gls_fit <- gls(
+  FEV1 ~ RACE + ARMCD * AVISIT, data = fev_data[complete.cases(fev_data), ],
+  correlation = nlme::corSymm(form = ~ VISITN | USUBJID),
+  weights = nlme::varIdent(form = ~ 1|VISITN)
+)
+gls_ests <- as.data.frame(
+  emmeans(
+    gls_fit,
+    spec = trt.vs.ctrl ~ ARMCD | AVISIT,
+    data = fev_data[complete.cases(fev_data), ],
+    weights = "proportional",
+    mode = "df.error"
+  )$contrasts
+)$estimate
+
+# compute marginal estimates for glmmTMB
+glmmTMB_fit <- glmmTMB(
+  FEV1 ~ RACE + ARMCD * AVISIT + us(0 + AVISIT | USUBJID),
+  data = fev_data,
+  dispformula = ~ 0
+)
+glmmTMB_ests <- as.data.frame(
+  emmeans(
+    glmmTMB_fit,
+    spec = trt.vs.ctrl ~ ARMCD | AVISIT,
+    weights = "proportional"
+  )$contrasts
+)$estimate
+
+## # compute marginal estimates from PROC MIXED
```

Comment:
Instead of commenting this code better use a condition flag and an if Block around it